swift/log/SwiftDiagnostics.h

```diff
@@ -16,16 +18,27 @@ extern const std::string_view programName;
 // Models a diagnostic source for Swift, holding static information that goes out into a diagnostic
 // These are internally stored into a map on id's. A specific error log can use binlog's category
 // as id, which will then be used to recover the diagnostic source while dumping.
-class SwiftDiagnosticsSource {
- public:
-  // creates a SwiftDiagnosticsSource with the given data
-  static void create(std::string_view id,
-                     std::string name,
-                     std::vector<std::string> helpLinks,
-                     std::string action);
+struct SwiftDiagnosticsSource {
+  std::string_view id;
+  std::string_view name;
+  static constexpr std::string_view extractorName = "swift";
+  std::string_view action;
+  std::string_view helpLinks;  // space separated if more than 1. Not a vector to allow constexpr
```

Comment:
`std::array` needs the `N` template argument, which then would prevent being able to use this type directly in the id mapping. This could be solved by introducing an abstract superclass (for example one with `virtual std::vector<std::string_view> getHelpLinks() = 0`), but I decided it was a bit too much of a fuss. Also, I expect help links to usually be just one (but not necessarily always, `internal_error` might be the one exception with two links for public and private issues).

The `TODO(C++20)` seems like a sensible solution.